jerry-core/vm/vm-stack.c

```diff
@@ -79,8 +80,11 @@ vm_stack_context_abort (vm_frame_ctx_t *frame_ctx_p, /**< frame context */
 #if ENABLED (JERRY_ES2015)
     case VM_CONTEXT_FOR_OF:
     {
+      ecma_value_t iterator = vm_stack_top_p[-3];
       ecma_free_value (vm_stack_top_p[-2]);
-      ecma_free_value (vm_stack_top_p[-3]);
+
+      ecma_op_iterator_close (iterator);
```

Comment:
What are the possible side effects?

Reply:
This operation may changes the `JERRY_CONTEXT (error_value)` and may applies the `ECMA_STATUS_EXCEPTION` status flags. At this point there is no way to handle exception, so the caller's responsibility to check whether the context abort raised error. See [the example](https://github.com/jerryscript-project/jerryscript/pull/3401/files#diff-4eae7d69004d99f61986f6e351bda007R3632) here.

Reply:
What happens if multiple iterators are closed? Nested `for-if`-s. Are they working correctly if their closes throws errors?

Reply:
The iterator close operation may raises error but it never overwrites the original error which caused the context abort. Each context abort is called, see the example [here](https://github.com/jerryscript-project/jerryscript/pull/3401/files#diff-bf7b2eb18cb71828704e5cc5bad6a567R211).